src/brogue/Items.c

```diff
@@ -3849,37 +3849,31 @@ void haste(creature *monst, short turns) {
     }
 }
 
+void healStatus(creature *monst, short status, short minStatus, short percent) {
+    short amount = max(1, percent * player.maxStatus[status] / 100);
+    short newStatus = monst->status[status] - amount;
+    monst->status[status] = min( monst->status[status], max( minStatus, newStatus ));
+}
+
 void heal(creature *monst, short percent, boolean panacea) {
     char buf[COLS], monstName[COLS];
-    monst->currentHP = min(monst->info.maxHP, monst->currentHP + percent * monst->info.maxHP / 100);
+    short amount = max(1, percent * monst->info.maxHP / 100);
+    monst->currentHP = min(monst->info.maxHP, monst->currentHP + amount);
     if (panacea) {
-        if (monst->status[STATUS_HALLUCINATING] > 1) {
-            monst->status[STATUS_HALLUCINATING] = 1;
-        }
-        if (monst->status[STATUS_CONFUSED] > 1) {
-            monst->status[STATUS_CONFUSED] = 1;
-        }
-        if (monst->status[STATUS_NAUSEOUS] > 1) {
-            monst->status[STATUS_NAUSEOUS] = 1;
-        }
-        if (monst->status[STATUS_SLOWED] > 1) {
-            monst->status[STATUS_SLOWED] = 1;
-        }
-        if (monst->status[STATUS_WEAKENED] > 1) {
-            monst->weaknessAmount = 0;
-            monst->status[STATUS_WEAKENED] = 0;
-            updateEncumbrance();
-        }
-        if (monst->status[STATUS_POISONED]) {
-            monst->poisonAmount = 0;
-            monst->status[STATUS_POISONED] = 0;
-        }
-        if (monst->status[STATUS_DARKNESS] > 0) {
-            monst->status[STATUS_DARKNESS] = 0;
-            if (monst == &player) {
-                updateMinersLightRadius();
-                updateVision(true);
-            }
+        healStatus( monst, STATUS_HALLUCINATING, 1, percent );
+        healStatus( monst, STATUS_CONFUSED,      1, percent );
+        healStatus( monst, STATUS_NAUSEOUS,      1, percent );
+        healStatus( monst, STATUS_SLOWED,        1, percent );
+        healStatus( monst, STATUS_WEAKENED,      0, percent );
+        healStatus( monst, STATUS_POISONED,      0, percent );
+        healStatus( monst, STATUS_DARKNESS,      0, percent );
```

Comment:
question: So, it seems the implementation here is to chop percentages off the status duration, as opposed to increasing the tick-down speed. Is this what we want? It means that to heal of all statuses, you can stand in a bloodwort cloud for as long as it takes you to heal to full HP. We can multiply the percentages down, but that still has healing be O(bloodworts) and not O(status duration).

Reply:
I agree that the code is mergable quality, but the gameplay consequences are "above my pay grade" - I don't feel like I have the level of play experience to judge that.

For testing I setup each of the statuses then wandered about for a little bit (to run the clock down), then healed with bloodwort.  The effect seemed reasonable and would definitely be helpful to a player in that situation.

I also played about 10 games on the branch to see how often the situation might crop up.  Roughly 3 times I could have sought out bloodwort for healing, but didn't really need to, and once I stood near a pod while I tested potions which proved beneficial.

So it seems the ultimate gameplay value is to recover from potion testing, or one of the long duration statuses like weakness, where the distance to the plant is small compared to the length of the buff.

Reply:
Thanks for the thorough explanation - will merge